fix(oncall): change labels schema from TypeList to TypeSet

sidharth-chauhan opened the pull request:
### What changed
- Updated the `labels` field in the `grafana_oncall_integration` resource schema from **`TypeList`** to **`TypeSet`**.
- Adjusted expand/flatten logic to correctly handle values with `(*schema.Set).List()`.
- Labels are now treated as a unique collection of `{ key, value }` pairs, rather than an ordered list.

### Why this is needed
- With `TypeList`, Terraform treated labels as **order-dependent**.  
  Example: swapping `{ key = "A", value = "1" }` and `{ key = "B", value = "2" }` triggered a diff, even though the data was the same.  
- This caused unnecessary plan/apply differences and did not reflect the intended behavior of OnCall labels.  
- `TypeSet` is the correct schema type for representing **unordered, unique mappings**, ensuring labels behave consistently.

### Benefits
- Prevents false diffs in `terraform plan` caused by label ordering.
- Matches Terraform best practices for representing sets of objects.
- Improves user experience when managing OnCall integrations with labels.

### Related Issue
Fixes: #2169

### Notes for reviewers
- Verified locally: provider builds successfully, `labels` appear in the plan output as expected, and no runtime errors occur.
- Full validation (e.g. ensuring no-diff behavior when reordering labels) will be covered by existing acceptance tests running in CI with Grafana Cloud.

sidharth-chauhan committed:
fix(oncall): change labels schema from TypeList to TypeSet (#2169)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/grafana/terraform-provider-grafana?pullRequest=2327) <br/>All committers have signed the CLA.

sidharth-chauhan committed:
docs: regenerate oncall_integration docs after changing labels schema

sidharth-chauhan commented:
Hi 👋 I’ve updated the schema from TypeList to TypeSet and regenerated the docs.
The failing checks are the Enterprise/Cloud acceptance tests which I understand don’t run on forked PRs. Please let me know if anything else is needed. Thanks!

sidharth-chauhan committed:
fix(oncall): switch labels to TypeSet and regenerate docs

sidharth-chauhan committed:
Fix oncall integration labels handling and update tests

matiasb commented:
Hi @sidharth-chauhan, thanks for the contribution. To be sure, it looks like the changes in the schema were lost?

sidharth-chauhan committed:
docs: regenerate after TypeSet change

sidharth-chauhan committed:
fixed schema

sidharth-chauhan committed:
fix(oncall): migrate labels to TypeSet and update tests

sidharth-chauhan commented:
Hi @matiasb, thanks for pointing that out 🙏. I’ve double-checked, and the schema changes (labels migrated to TypeSet) are present along with the updated tests. Please let me know if I missed something.

matiasb commented:
Hi, I was trying this and it seems it may not be backwards compatible? E.g. using the example from the [docs](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/oncall_integration), I'm able to set/change labels, but that same example fails with the updated provider:

```
│ Error: Unsupported argument
│ 
│   on example.tf line 78, in resource "grafana_oncall_integration" "test-acc-integration":
│   78:   labels         = [data.grafana_oncall_label.test-label, data.grafana_oncall_label.test-another-label]
│ 
│ An argument named "labels" is not expected here. Did you mean to define a block of type "labels"?
╵
╷
│ Error: Unsupported argument
│ 
│   on example.tf line 79, in resource "grafana_oncall_integration" "test-acc-integration":
│   79:   dynamic_labels = [data.grafana_oncall_label.test-dynamic-label]
│ 
│ An argument named "dynamic_labels" is not expected here. Did you mean to define a block of type
│ "dynamic_labels"?
```

sidharth-chauhan committed:
oncall: flatten labels to only key/value (remove id field) to match schema/tests

sidharth-chauhan committed:
fix(oncall): migrate labels and dynamic_labels schema to TypeSet of TypeMap

sidharth-chauhan commented:
Hi @matiasb, I’ve updated the `grafana_oncall_integration` resource to migrate labels from TypeList to TypeSet 
and adjusted expand/flatten logic to accept both TypeSet/TypeMap element shapes for compatibility. 

Local builds and package tests pass. Could a maintainer please trigger the full acceptance tests 
and let me know if any further adjustments are needed? Thanks!

sidharth-chauhan committed:
Merge branch 'grafana:main' into fix-oncall-labels-2169

sidharth-chauhan committed:
Merge branch 'main' into fix-oncall-labels-2169
